modules/page-memory/src/main/java/org/apache/ignite/internal/pagememory/persistence/io/PartitionMetaIo.java

```diff
@@ -19,213 +19,32 @@
 
 import static org.apache.ignite.internal.pagememory.PageIdAllocator.FLAG_AUX;
 import static org.apache.ignite.internal.pagememory.util.PageUtils.getInt;
-import static org.apache.ignite.internal.pagememory.util.PageUtils.getLong;
 import static org.apache.ignite.internal.pagememory.util.PageUtils.putInt;
-import static org.apache.ignite.internal.pagememory.util.PageUtils.putLong;
 
-import org.apache.ignite.internal.hlc.HybridTimestamp;
-import org.apache.ignite.internal.lang.IgniteStringBuilder;
-import org.apache.ignite.internal.pagememory.io.IoVersions;
 import org.apache.ignite.internal.pagememory.io.PageIo;
 
 /**
- * Io for partition metadata pages.
+ * Base Io for partition metadata pages.
  */
-public class PartitionMetaIo extends PageIo {
-    private static final int LAST_APPLIED_INDEX_OFF = COMMON_HEADER_END;
-
-    private static final int LAST_APPLIED_TERM_OFF = LAST_APPLIED_INDEX_OFF + Long.BYTES;
-
-    private static final int LAST_REPLICATION_PROTOCOL_GROUP_CONFIG_FIRST_PAGE_ID_OFF = LAST_APPLIED_TERM_OFF + Long.BYTES;
-
-    private static final int FREE_LIST_ROOT_PAGE_ID_OFF = LAST_REPLICATION_PROTOCOL_GROUP_CONFIG_FIRST_PAGE_ID_OFF + Long.BYTES;
-
-    private static final int VERSION_CHAIN_TREE_ROOT_PAGE_ID_OFF = FREE_LIST_ROOT_PAGE_ID_OFF + Long.BYTES;
-
-    public static final int INDEX_TREE_META_PAGE_ID_OFF = VERSION_CHAIN_TREE_ROOT_PAGE_ID_OFF + Long.BYTES;
-
-    private static final int GC_QUEUE_META_PAGE_ID_OFF = INDEX_TREE_META_PAGE_ID_OFF + Long.BYTES;
-
-    private static final int PAGE_COUNT_OFF = GC_QUEUE_META_PAGE_ID_OFF + Long.BYTES;
-
-    private static final int LEASE_START_TIME_OFF = PAGE_COUNT_OFF + Integer.BYTES;
-
-    /** Page IO type. */
-    public static final short T_TABLE_PARTITION_META_IO = 7;
-
-    /** I/O versions. */
-    public static final IoVersions<PartitionMetaIo> VERSIONS = new IoVersions<>(new PartitionMetaIo(1));
+public abstract class PartitionMetaIo extends PageIo {
+    private final int getPageCountOffset;
 
     /**
      * Constructor.
      *
      * @param ver Page format version.
      */
-    protected PartitionMetaIo(int ver) {
-        super(T_TABLE_PARTITION_META_IO, ver, FLAG_AUX);
+    protected PartitionMetaIo(int type, int ver, int getPageCountOffset) {
+        super(type, ver, FLAG_AUX);
+        this.getPageCountOffset = getPageCountOffset;
```

Comment:
Why can't it be static in the beginning of the payload? You are allowed to fix the order of values. Can you please change that?